File: haskell/p24.hs
-- P24 Lotto: Draw N different random numbers from the set 1..M.

import System.Random
import Data.List

-- The obvious answer is P23.f1 n [1..m]
-- Let's use "nub" for a different solution
f1 :: RandomGen g => Int -> Int -> g -> [Int]
f1 n m = take n . nub . randomRs (1, m)

File: clojure/P24.clj
; P24 Lotto: Draw N different random numbers from the set 1..M.

; The obvious answer is (P23/f1 n (range 1 m))
; Let's use "distinct" for a different solution
(defn f1 [n m]
  (take n (distinct (repeatedly #(inc (rand-int m))))))